### PR TITLE
Update development tools

### PR DESCRIPTION
**Pull request by Bozydarek**
- migrate to docker compose v2 (now it's a plugin to docker)
- update README
- add types annotation and format code of dev script

Note: this should probably wait until the end of this year's edition

### CODE REVIEW

**Comment by ref-humbold** [comment]
> Note: this should probably wait until the end of this year's edition

Or you could retarget this PR to `develop` branch - it was created for this ;)

File: README.md [hunk 1]
```diff
@@ -24,18 +24,19 @@ production one.
 
 #### Required software
 
-To run docker container you obviously need to have _Docker (Community Edition)_ installed.
+To run container for development you need to have _Docker (Community Edition)_ installed, as well as
+a `docker-compose-plugin` installed on your system.
 
-Additionally, you should install `docker-compose` using APT on Debian-based systems:
+For Debain-based systems use APT:
```
Comment (ref-humbold) [inline README.md:30]: Typo: `Debian` :P

File: tools/dev.py [hunk 2]
```diff
@@ -21,205 +24,256 @@ class Colour:
...
-elif args.command in ["shell", "sh"]:
-    if args.shell in ["bash"]:
-        docker_exec(["/bin/bash"], WEB_CONTAINER_NAME)
-    elif args.shell in ["postgres", "psql"]:
-        docker_exec(["psql", "-U", "zosia"], DB_CONTAINER_NAME)
-    else:
-        shell_parser.print_help()
-elif args.command in ["migrations", "m"]:
-    if args.action in ["apply", "a"]:
+def cli():
```
Comment (ref-humbold) [inline tools/dev.py:105]: If you'd like to enclose it in a function, you should split the new function into parser generation and argument deduction.
```
def cli():
    parser = create_parser()
    args = parser.parse_args()
    deduce_args(args)
```

Comment (Bozydarek) [reply tools/dev.py:105]: Splitting it as proposed has one major issue - we would have to return multiple parsers in order to handle help printing for subparsers :confused: 

```
    else:
          migrate_parser.print_help()
```

File: README.md [hunk 3]
```diff
@@ -49,7 +50,8 @@ There are 3 flags that you might want to use with `dev.py run`:
 
 - `--create-admin` - create super user / admin account with the password of your choice
 - `--create-data` - generate some semi-random data - the implementation can be found
-  here: `src/conferences/management/commands/create_data.py`
+  here: `src/conferences/management/commands/create_data.py`. This command also generates
+  several user accounts `zosia[0-9]?@example.com` that can be accessed with password: `pass`.
```
Comment (ref-humbold) [inline README.md:54]: > several user accounts `zosia[0-9]?@example.com`

It's currently 5 normal accounts `zosia[1-5]@example.com` and one staff account `zosia@example.com`.

**Review by ref-humbold — APPROVED** [review]

**Pull request merged**